deploy/001_deploy_swap.ts

```diff
@@ -0,0 +1,54 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { DeployFunction } from "hardhat-deploy/types";
+import {
+  getProxyAdminAddress,
+  getSideWrappedBtcAddress,
+} from "../utils/address";
+import { ethers } from "hardhat";
+import { DeployResult } from "hardhat-deploy/dist/types";
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts } = hre;
+  const { deploy } = deployments;
+
+  const { deployer } = await getNamedAccounts();
+  const proxyAdminAddress = await getProxyAdminAddress(hre);
+  const sideWrappedBtcAddress = await getSideWrappedBtcAddress(hre);
+
+  const factorySwapRBTC = await ethers.getContractFactory(
+    "SwapRBTC",
+    proxyAdminAddress
+  );
+
+  const deploySwap: DeployResult = await deploy('SwapRBTC', {
+    from: deployer,
+    log: true
+  });
+
+  const swapInitializeParams = [sideWrappedBtcAddress];
+  const swapInstance = factorySwapRBTC.attach(deploySwap.address);
```

Comment:
actually swapRbtc does not have a factory

Reply:
it`s actually not a factory like Contract Factory, it is a factory of multiple contracts like ethers.ContractFactory